platform-dom/src/main/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompleteIdMapFactory.java

```diff
@@ -0,0 +1,51 @@
+package com.softicar.platform.dom.elements.input.auto;
+
+import com.softicar.platform.common.core.entity.EntityIdFromDisplayStringExtractor;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+
+/**
+ * Attempts to extract numeric identifiers (IDs) from the displayed strings of
+ * auto-complete elements.
+ * <p>
+ * ID extraction is performed using {@link EntityIdFromDisplayStringExtractor}.
+ * If ID extraction for one or more elements fails, or if the same ID is
+ * extracted for different elements, the whole ID extraction is considered to
+ * have failed.
+ *
+ * @author Oliver Richers
+ */
+class DomAutoCompleteIdMapFactory<T> {
+
+	private final Map<String, T> stringToElementMap;
+
+	public DomAutoCompleteIdMapFactory(Map<String, T> stringToElementMap) {
+
+		this.stringToElementMap = stringToElementMap;
```

Comment:
```suggestion
		this.stringToElementMap = Objects.requireNonNull(stringToElementMap);
```
we should probably fail early here

Reply:
did so